Update the Rails.start API (#905)
elastic/apm-agent-ruby: docs/api.asciidoc

[[rails-start]]
=== Rails

> Start the agent and hook into Rails manually. If you skip requiring the gem,
> you won't have the `Railtie` and would have to define your own initializer.
> Note that this is not the typical way to setup and start the agent in a Rails app.

Reviewer: ```suggestion
Start the agent and hook into Rails manually. If you skip requiring the gem,
you will need to start the agent yourself in for example an initializer.
Note that this is not the typical way to setup and start the agent in a Rails app.
```

[source,ruby]
----

> ElasticAPM::Rails.start(MyRailsApp.new, server_url: 'http://localhost:8200')

Reviewer: Aren't there a bunch of side effects from calling `new` on an app?

Reviewer: I'd be a little bit concerned about `new`ing a rails app. Too much magic going on 😅

Author: If a user doesn't use a Railtie, then how do they initialize a Rails app? A user would pass their app instance to the `ElasticAPM::Rails.start` method and however they instantiate it is up to them. Do you know of a better example of instantiating a rails app without a Railtie?

Reviewer: I might be misunderstanding, but I don't think it's our job to start nor instantiate users' apps?

If someone wants to use this API, I think they should do so in an initializer or their own custom railtie. And when they do, they'll have the application object at hand. I don't think we ought to encourage any other way to start the agent with Rails.

Author: Yes, we are saying the same thing. I was just trying to provide a simple example. So would this be better? The `myapp` object is whatever application object they have in an initializer or custom railtie.

`ElasticAPM::Rails.start(myapp, server_url: 'http://localhost:8200')`

Reviewer: I think so. Or maybe include a whole example of starting in an initializer (including a note that this is the hard way)

Author: ok I added an example initializer
